packages/plugin-assets-retry/src/runtime/asyncChunkRetry.ts

```diff
@@ -95,18 +95,33 @@ function getUrlRetryQuery(
   return '';
 }
 
+// "http://cdn.com/app/main/static/js/index.js" -> "/app/main/static/js/index.js"
+function getPathFromUrl(url: string): string {
+  const protocolStartIndex = url.indexOf('//');
+
+  // case /app/main/static/js/index.js
+  if (protocolStartIndex === -1 && url.startsWith('/')) {
+    return url;
+  }
+
+  // case "//cdn.com/app/main/static/js/index.js"
+  // case "http://cdn.com/app/main/static/js/index.js"
+  const protocolEndIndex = protocolStartIndex + 2;
+  const pathStartIndex = url.indexOf('/', protocolEndIndex);
+
+  const path = url.slice(pathStartIndex);
+  return path;
+}
+
 function getNextRetryUrl(
   existRetryTimes: number,
   nextDomain: string,
   originalSrcUrl: string,
-  originalScriptFilename: string,
 ) {
+  const originalSrcUrlWithoutDomain = getPathFromUrl(originalSrcUrl);
```

Comment:
Can we just name it `pathname`? similar to `location.pathname`.

```suggestion
  const originalPathname = getPathFromUrl(originalSrcUrl);
```

Reply:
In fact, this also has a query , I rename the `getPathFromUrl` to `removeDomainFromUrl`

Reply:
How about `getRelativeUrl()`

![image](https://github.com/web-infra-dev/rsbuild/assets/7237365/e6015141-a57b-4a9d-919e-04a10cebd889)

Reply:
🙂 ... I don't think "relativeURL" is good enough, and a little bit confusing, this variable name will be minified by terser, so it doesn't matter

Reply:
The variable name matters because the current name is too long and unreadable.

According to the [URI RFC](https://www.rfc-editor.org/rfc/rfc3986#section-4.2), we can call it `absolutePath`.

![image](https://github.com/web-infra-dev/rsbuild/assets/7237365/4f012e4f-0ca4-45cf-8e5c-ee323b0068e5)

Reply:
yeah, `relativeUrl` is consistent with `absolutePath`

Reply:
no, a little different... fixed